pkg/pebinaryexecutor/pebinary.go

```diff
@@ -160,9 +165,9 @@ func (p *PEBinaryConfig) Kubelet(ctx context.Context, args []string) error {
 		cleanArgs = append(cleanArgs, arg)
 	}
 
-	logrus.Infof("Running RKE2 kubelet %v", cleanArgs)
-	go func() {
+	win.ProcessWaitGroup.StartWithContext(ctx, func(ctx context.Context) {
```

Comment:
The initial implementation of `pebinary` had access to this context as well. iiuc, using a separate context ensures that the CNI plugin is restarted each time that the kubelet is restarted, which may happen before the global context is canceled (for example, if someone accidentally runs  `Stop-Process` for the kubelet). Since this PR does not focus on how the kubelet starts and stops the CNI plugins I opted to leave this unchanged.